keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/CheckScheduler.kt

```diff
@@ -26,7 +27,8 @@ class CheckScheduler(
   private val environmentPromotionChecker: EnvironmentPromotionChecker,
   @Value("\${keel.resource-check.min-age-duration:60s}") private val resourceCheckMinAgeDuration: Duration,
   @Value("\${keel.resource-check.batch-size:1}") private val resourceCheckBatchSize: Int,
-  private val publisher: ApplicationEventPublisher
+  private val publisher: ApplicationEventPublisher,
+  private val agentLockRepository: AgentLockRepository
```

Comment:
I think that `enabled` (below on line 35) should be an `AtomicBoolean` to make sure that this is thread safe.

Reply:
make sense. I haven't seen any usage of `AtomicBoolean` in keel, but I'll add it here.

Reply:
Qe do it in other services around locking - like [here](https://github.com/spinnaker/swabbie/blob/master/swabbie-core/src/main/kotlin/com/netflix/spinnaker/swabbie/discovery/DiscoveryActivated.kt#L14)